Libraries/Components/ScrollView/ScrollViewStickyHeader.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule ScrollViewStickyHeader
+ * @flow
+ */
+'use strict';
+
+const Animated = require('Animated');
+const React = require('React');
+const StyleSheet = require('StyleSheet');
+
+type Props = {
+  children?: React.Element<*>,
+  scrollAnimatedValue: Animated.Value,
+  onLayout: (event: Object) => void,
+};
+
+class ScrollViewStickyHeader extends React.Component {
+  props: Props;
+  state = {
+    measured: false,
+    layoutY: 0,
+    nextHeaderLayoutY: (null: ?number),
+  };
+
+  setNextHeaderY(y: number) {
+    this.setState({ nextHeaderLayoutY: y });
+  }
+
+  _onLayout = (event) => {
+    this.setState({
+      measured: true,
+      layoutY: event.nativeEvent.layout.y,
+    });
+
+    this.props.onLayout(event);
+  };
+
+  render() {
+    const {measured, layoutY, nextHeaderLayoutY} = this.state;
+
+    let translateY;
+    if (measured) {
+      const inputRange = [-1, 0, layoutY];
+      const outputRange: Array<number> = [-1, 0, 0];
+      if (nextHeaderLayoutY != null) {
+        inputRange.push(nextHeaderLayoutY, nextHeaderLayoutY + 1);
+        outputRange.push(nextHeaderLayoutY - layoutY, nextHeaderLayoutY - layoutY);
+      } else {
+        inputRange.push(layoutY + 1);
+        outputRange.push(1);
+      }
+      translateY = this.props.scrollAnimatedValue.interpolate({
+        inputRange,
+        outputRange,
+      });
+    } else {
+      translateY = 0;
+    }
+
+    const child = React.Children.only(this.props.children);
+
+    return (
+      <Animated.View
+        collapsable={false}
+        onLayout={this._onLayout}
+        style={[child.props.style, styles.header, {transform: [{translateY}]}]}>
+        {React.cloneElement(child, {
+          style: styles.fill,
+        })}
+      </Animated.View>
+    );
+  }
+}
+
+const styles = StyleSheet.create({
+  header: {
+    zIndex: 10,
```

Comment:
I don't really have other ideas except maybe exporting this as a constant somewhere.